### PR TITLE
Use Puppet 4.x instead of 3.4.x

### PR DESCRIPTION
**Pull request by jerith**

### CODE REVIEW

**Comment by jerith** [comment]
This defaults to Puppet 3.8.x, so we should probably change that perhaps remove the optional 3.8.x support before merging.

File: puppet/puppet-bootstrap.sh [hunk 1]
```diff
@@ -1,9 +1,39 @@
...
+
+# Upgrade to new Puppet and install some dependencies.
+${puppet_remove}
+${puppet_install}
+apt-get autoremove -qy
 apt-get install -qy --no-install-recommends bundler git ruby-dev
 
-# Install gems
-bundle install --without test
+# If we're using Puppet 4.x, symlink it into /usr/sbin because sudo and $PATH.
```
Comment (JayH5) [inline puppet/puppet-bootstrap.sh:36]: `/usr/sbin` or `/sbin`?

Comment (jerith) [inline puppet/puppet-bootstrap.sh:36]: Whoops. The comment is correct, the code is wrong. Good catch.

**Comment by JayH5** [comment]
:+1:

**Comment by JayH5** [comment]
Maybe worth waiting for the seed_stack module to be verified/tested on Puppet 4 (although it doesn't really have many tests yet)?

**Comment by JayH5** [comment]
:+1:

**Comment by JayH5** [comment]
:+1:

**Pull request merged**